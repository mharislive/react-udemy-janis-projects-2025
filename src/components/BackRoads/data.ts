import { faSocks, faTree, faWallet } from "@fortawesome/free-solid-svg-icons";
import tour1 from "../../assets/backroads/tour-1.jpeg";
import tour2 from "../../assets/backroads/tour-2.jpeg";
import tour3 from "../../assets/backroads/tour-3.jpeg";
import tour4 from "../../assets/backroads/tour-4.jpeg";

export const serviceData = [
  {
    title: "saving money",
    content:
      "Lorem ipsum dolor sit amet consectetur adipisicing elit. Asperiores, officia.",
    icon: faWallet,
  },
  {
    title: "endless hiking",
    content:
      "Lorem ipsum dolor sit amet consectetur adipisicing elit. Asperiores, officia.",
    icon: faTree,
  },
  {
    title: "amazing comfort",
    content:
      "Lorem ipsum dolor sit amet consectetur adipisicing elit. Asperiores, officia.",
    icon: faSocks,
  },
];

export const toursdata = [
  {
    content:
      "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Cumque vitae tempore voluptatum maxime reprehenderit eum quod exercitationem fugit, qui corporis.",
    image: tour1,
    date: "august 26th, 2020",
    title: "Tibet Adventure",
    country: "china",
    duration: "6 days",
    price: "$2100",
  },
  {
    content:
      "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Cumque vitae tempore voluptatum maxime reprehenderit eum quod exercitationem fugit, qui corporis.",
    image: tour2,
    date: "october 1th, 2020",
    title: "best of java",
    country: "indonesia",
    duration: "11 days",
    price: "$1400",
  },
  {
    content:
      "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Cumque vitae tempore voluptatum maxime reprehenderit eum quod exercitationem fugit, qui corporis.",
    image: tour3,
    date: "september 15th, 2020",
    title: "explore hong kong",
    country: "hong kong",
    duration: "8 days",
    price: "$5000",
  },
  {
    content:
      "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Cumque vitae tempore voluptatum maxime reprehenderit eum quod exercitationem fugit, qui corporis.",
    image: tour4,
    date: "december 5th, 2019",
    title: "kenya highlights",
    country: "kenya",
    duration: "20 days",
    price: "$3300",
  },
];
