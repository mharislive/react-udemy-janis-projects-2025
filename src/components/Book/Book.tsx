import type { BookType } from "../../types/Book";

const Book = (props: BookType) => {
  const { title, author, image } = props;

  return (
    <article className="p-5 rounded-lg border-1">
      <img src={image} alt={title} className="mx-auto mb-4 max-w-full" />
      <h2 className="text-2xl mb-1">{title}</h2>
      <h3 className="text-sm">By {author}</h3>
    </article>
  );
};

export default Book;
