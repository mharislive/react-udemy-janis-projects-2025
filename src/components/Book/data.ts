import type { BookType } from "../../types/Book";

export const BOOKS: BookType[] = [
  {
    title: "Atomic Habits",
    author: "James Clear",
    image: "https://m.media-amazon.com/images/I/81F90H7hnML._SY425_.jpg",
    id: 1,
  },
  {
    title: "Word Power Made Easy",
    author: "Norman Lewis",
    image: "https://m.media-amazon.com/images/I/71DortY2MXL._SY425_.jpg",
    id: 2,
  },
  {
    title: "The Psychology of Money Paperback",
    author: "Morgan Housel",
    image: "https://m.media-amazon.com/images/I/71XEsXS5RlL._SY425_.jpg",
    id: 3,
  },
];
