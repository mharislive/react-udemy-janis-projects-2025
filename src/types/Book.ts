export type BookType = {
  author: string;
  title: string;
  image: string;
  id: number;
};
