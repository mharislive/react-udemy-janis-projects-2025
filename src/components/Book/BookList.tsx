import Book from "../Book/Book";
import { BOOKS } from "./data";

const BookList = () => {
  return (
    <section className="grid grid-cols-1 md:grid-cols-3 gap-4 text-center">
      {BOOKS.map((book) => (
        <Book key={book.id} {...book} />
      ))}
    </section>
  );
};

export default BookList;
