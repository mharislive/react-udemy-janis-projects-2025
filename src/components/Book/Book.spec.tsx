import { render, screen } from "@testing-library/react";
import { axe } from "vitest-axe";
import Book from "./Book";
import { BOOKS } from "./data";

describe("Book", () => {
  it("should render successfully", () => {
    render(<Book {...BOOKS[0]} />);
  });

  it("should render title", () => {
    render(<Book {...BOOKS[0]} />);
    const title = screen.getByRole("heading", { level: 2 });
    expect(title).toBeInTheDocument();
    expect(title).toHaveTextContent("Atomic Habits");
  });

  it("should render author name", () => {
    render(<Book {...BOOKS[0]} />);
    const author = screen.getByRole("heading", { level: 3 });
    expect(author).toBeInTheDocument();
    expect(author).toHaveTextContent("James Clear");
  });

  it("should render the image", () => {
    render(<Book {...BOOKS[0]} />);
    const image = screen.getByRole("img");
    expect(image).toBeInTheDocument();
    expect(image).toHaveAttribute("alt", "Atomic Habits");
  });

  it("should match the snapshot", () => {
    const { asFragment } = render(<Book {...BOOKS[0]} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should have no accessibility violations", async () => {
    const { container } = render(<Book {...BOOKS[0]} />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
