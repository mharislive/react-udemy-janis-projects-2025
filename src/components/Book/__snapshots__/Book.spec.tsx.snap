// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Book > should match the snapshot 1`] = `
<DocumentFragment>
  <article
    class="p-5 rounded-lg border-1"
  >
    <img
      alt="Atomic Habits"
      class="mx-auto mb-4 max-w-full"
      src="https://m.media-amazon.com/images/I/81F90H7hnML._SY425_.jpg"
    />
    <h2
      class="text-2xl mb-1"
    >
      Atomic Habits
    </h2>
    <h3
      class="text-sm"
    >
      By James Clear
    </h3>
  </article>
</DocumentFragment>
`;
